modules/components/pages/outputs/azure_data_lake_gen2.adoc

```diff
@@ -0,0 +1,123 @@
+= azure_data_lake_gen2
+//tag::single-source[]
+:type: output
+:page-beta: true
+
+:categories: ["Services","Azure"]
+
+
+// © 2024 Redpanda Data Inc.
+
+
+component_type_dropdown::[]
+
+
+Sends message parts as files to an https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction[Azure Data Lake Gen2^] file system. Each file is uploaded with the file name specified in the `path` field.
+
+ifndef::env-cloud[]
+Introduced in version 4.38.0.
+endif::[]
+
+
+```yml
+# Configuration fields, showing default values
+output:
+  label: ""
+  azure_data_lake_gen2:
+    storage_account: "" # No default (optional)
+    storage_access_key: "" # No default (optional)
+    storage_connection_string: "" # No default (optional)
+    storage_sas_token: "" # No default (optional)
+    filesystem: messages-${!timestamp("2006")} # No default (required)
+    path: ${!counter()}-${!timestamp_unix_nano()}.txt
+    max_in_flight: 64
+```
+To specify a different <<path,`path` value>> (file name) for each file, use xref:configuration:interpolation.adoc#bloblang-queries[function
+interpolations]. Function interpolations are calculated for each message in a batch.
+
+== Authentication methods
+
+This output supports multiple authentication methods. You must configure at least one method from the following list:
+
+- `storage_connection_string`
+- `storage_account` and `storage_access_key`
+- `storage_account` and `storage_sas_token`
+- `storage_account` to access using https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#DefaultAzureCredential[DefaultAzureCredential^]
+
+If you configure multiple authentication methods, the `storage_connection_string` takes precedence.
+
+== Performance
+
+This output benefits from sending multiple messages in flight in parallel for improved performance. You can tune the max number of in flight messages (or message batches) with the field `max_in_flight`.
+
+== Fields
+
+=== `storage_account`
```

Comment:
Could we give examples to show the format of the value that is expected?

Reply:
The values are copied 1-to-1 from Azure, i.e. there isn't a specific format the user would have to synthesize the value into, so I don't know if examples would be that useful. For what it's worth, it is the exact same set of fields used by [the `azure_blob_storage` plugin](https://docs.redpanda.com/redpanda-connect/components/inputs/azure_blob_storage/).